### PR TITLE
Make lexicals (alias, import and require) return the given module(s) instead of nil

### PR DESCRIPTION
**Pull request by aesedepece**
Implements changes suggested by @josevalim at #4596.

```elixir
iex> import String
String
```

```elixir
iex> require String
String
```

```elixir
iex> alias String
String
```

```elixir
iex> alias String, as: Foo
String
iex> alias Foo, as: Bar
String
```

```elixir
iex> alias String.{Break, Casing, Chars}
[String.Break, String.Casing, String.Chars]
```

Little updates to section 13 of the [Getting Started](https://github.com/elixir-lang/elixir-lang.github.com/blob/master/getting-started/alias-require-and-import.markdown) may be required, so I will take care of it upon merge.

This is my very first PR to this project, but I am really eager to keep contributing.

We are using Elixir extensively at [Stampery](https://github.com/stampery), so we will try to give back as much as possible to this community. :smiley_cat:

### CODE REVIEW

**Comment by aesedepece** [comment]
This change enables this ugly yet classical construction for 1-time imports:

```elixir
(import Foo).bar()
```

File: lib/elixir/src/elixir_translator.erl [hunk 1]
```diff
@@ -50,8 +50,8 @@ translate({{'.', _, [erlang, 'orelse']}, Meta, [Left, Right]}, S) ->
 
 %% Lexical
 
-translate({Lexical, _, [_, _]}, S) when Lexical == import; Lexical == alias; Lexical == require ->
-  {{atom, 0, nil}, S};
+translate({Lexical, _, [Modules, _]}, S) when Lexical == import; Lexical == alias; Lexical == require ->
```
Comment (lexmag) [inline lib/elixir/src/elixir_translator.erl]: Let's use singular `Module` here. :bowtie:

**Comment by antipax** [comment]
@aesedepece I'm pretty sure the import will still be in scope for all lines after the import in the current scope? So it might not be a one time import, but in any event the point of this was just to look good in iex when you import stuff, so good work! :)

**Comment by whatyouhide** [comment]
I think it would be fair to ping @tuvistavie as he mentioned he was working on this, so he doesn't do the work for nothing? :)

**Comment by aesedepece** [comment]
@lexmag Thanks for the suggestion. I used "Modules" just because it returns a List with several modules for the `alias Foo.{bar, ber, bir}` case. Now I have edited it → "Module".

@antipax 100% right, the import will remain in scope, the 1-time thing was just a curiosity.

@whatyouhide I'm sorry I took it without noticing he had already claimed it :sweat_smile:

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by eksperimental** [comment]
@aesedepece it will be good if you could ad as tests the very same usage examples you have used above

**Comment by josevalim** [comment]
I just pushed assertions to import_test. If someone wants to add more
comprehensive tests, please go ahead.

On Sunday, May 15, 2016, eksperimental <notifications@github.com> wrote:

> @aesedepece <https://github.com/aesedepece> it will be good if you could
> ad as tests the very same usage examples you have used above
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4629#issuecomment-219294042>
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by lexmag** [comment]
Extra tests for alias, import and require are in master.

**Comment by eksperimental** [comment]
Sweet

**Comment by tuvistavie** [comment]
@whatyouhide @aesedepece Thanks for the ping, and no problem, I did not have a chance to take a look at it yet!

**Pull request merged**